lib/cell.rb

```diff
@@ -0,0 +1,63 @@
+class Cell
+
+  attr_reader :mine, :cleared, :flagged, :adjacent_mines
+
+  def initialize
+    @mine = false
+    @cleared = false
+    @flagged = false
+    @adjacent_mines = 0
+  end
+
+  # This ends the game!
+  def exploded?
+    mine && cleared
+  end
+
+  def place_mine
+    @mine = true
+  end
+
+  def count_adjacent_mine
+    @adjacent_mines += 1
+  end
+
+  def clear
+    @cleared = true unless flagged
+  end
+
+  def flag
+    @flagged = true unless cleared
+  end
+
+  def unflag
+    @flagged = false
```

Comment:
So I wonder if the user has a flag/unflag option, or if there's a flag option that in this case you will have to check if it's flagged or not and apply a different method if so?